src/modules/launcher/Plugins/Microsoft.Plugin.Program/Programs/Win32.cs

```diff
@@ -173,6 +201,69 @@ private static Win32 Win32Program(string path)
             }
         }
 
+        // This function filters Internet Shortcut programs
+        private static Win32 InternetShortcutProgram(string path)
+        {
+            string[] lines = System.IO.File.ReadAllLines(path);
+            string appName = string.Empty;
+            string iconPath = string.Empty;
+            string scheme = string.Empty;
+            bool validApp = false;
+
+            const string steamScheme = "steam";

```

Comment:
Shouldn't we be checking that it is not a web link rather than checking if it's `steam`? That way it would work for all applications which use internet based shortcuts rather than just steam. We could filter out `http`, `https` and any other web-based protocols.

Reply:
Yea that would be ideal to do but apparently there is no way to tell if an internet shortcut is pointing to a web page or an application. I did spend quite some time looking at it but did not find anything. These are the list of all the official schemes (https://en.wikipedia.org/wiki/List_of_URI_schemes) and they contain easily more than a hundred schemes. I did a search for how many actually return application shortcuts and seems like steam is the only one. I thought it was better to add steam than to filter out a hundred other schemes like `http`, `https` etc. Pls let me know if you have more context on filtering app based shortcuts. I would really appreciate it!

Reply:
Agreed, looking at that page there seems to be way too many. I wonder if there is some other attribute which actually distinguishes it. Do we want to limit steam based urls only to the ones which are `rungameid` i.e for launching games, or any of the commands (so a user may make a url shortcut to do something else instead)? This has a list of all of steam's protocols https://developer.valvesoftware.com/wiki/Steam_browser_protocol

Reply:
Thanks for sharing that, I wasn't aware of it. Yes, I think we would want to filter only those urls which actually run apps.

Reply:
Updated it to check if the hostname is `run` or `rungameid` as other hostnames do not launch apps. Also did not rather make the code see if it starts with the word `run` as there is another command, `runsafe` which resets variables of a game.